aws/container-linux/kubernetes/bastion.tf

```diff
@@ -0,0 +1,166 @@
+resource "aws_autoscaling_group" "bastion" {
+  name = "${var.cluster_name}-bastion ${aws_launch_configuration.bastion.name}"
+
+  # count
+  desired_capacity          = "${var.bastion_count}"
+  min_size                  = "${var.bastion_count}"
+  max_size                  = "${var.bastion_count}"
+  default_cooldown          = 30
+  health_check_grace_period = 30
+
+  # network
+  vpc_zone_identifier = ["${aws_subnet.private.*.id}"]
+
+  # template
+  launch_configuration = "${aws_launch_configuration.bastion.name}"
+
+  # target groups to which instances should be added
+  target_group_arns = [
+    "${aws_lb_target_group.bastion.id}"
+  ]
+
+  lifecycle {
+    # override the default destroy and replace update behavior
+    create_before_destroy = true
```

Comment:
[`create_before_destroy`](https://www.terraform.io/docs/configuration/resources.html#create_before_destroy) overrides the default terraform lifecycle for resource replacement. Normally terraform will destroy the old resource and then create a new one. Setting that attribute means it will instead launch the replacement resource before destroying the previous one.

In this case, a change requiring replacement of the ASG will be rolled out by creating a new ASG and then destroying the old one once the new one is created.